### DISCUSSION TITLE
Can't build due to ndkBuild error[CXX1405]

### DISCUSSION THREAD

**Discussion by wuxuan2811** [shadowsocks/shadowsocks-android#2833] [Q&A]
Got the latest version to build, but error below:
[CXX1405] error when building with ndkBuild using D:\Android\TestProjects\Git\shadowsocks-android\core\src\main\jni\Android.mk: Build command failed.
Error while executing process D:\Android\Sdk\ndk\21.4.7075529\ndk-build.cmd with arguments {NDK_PROJECT_PATH=null APP_BUILD_SCRIPT=D:\Android\TestProjects\Git\shadowsocks-android\core\src\main\jni\Android.mk NDK_APPLICATION_MK=D:\Android\TestProjects\Git\shadowsocks-android\core\src\main\jni\Application.mk APP_ABI=x86 NDK_ALL_ABIS=x86 NDK_DEBUG=1 APP_PLATFORM=android-23 NDK_OUT=D:\Android\TestProjects\Git\shadowsocks-android\core\build\intermediates\cxx\Debug\202y2hp6/obj NDK_LIBS_OUT=D:\Android\TestProjects\Git\shadowsocks-android\core\build\intermediates\cxx\Debug\202y2hp6/lib -j12 APP_SHORT_COMMANDS=false LOCAL_SHORT_COMMANDS=false -B -n}
md "D:\Android\TestProjects\Git\shadowsocks-android\core\build\intermediates\cxx\Debug\202y2hp6\lib\x86" >NUL 2>NUL || rem
D:/Android/Sdk/ndk/21.4.7075529/build//../prebuilt/windows-x86_64/bin/echo.exe [x86] "Gdbserver      ": "[i686-linux-android] D:\Android\TestProjects\Git\shadowsocks-android\core\build\intermediates\cxx\Debug\202y2hp6/lib/x86/gdbserver"
copy /b/y "D:\Android\Sdk\ndk\21.4.7075529\build\\..\prebuilt\android-x86\gdbserver\gdbserver" "D:\Android\TestProjects\Git\shadowsocks-android\core\build\intermediates\cxx\Debug\202y2hp6\lib\x86\gdbserver" > NUL
D:/Android/Sdk/ndk/21.4.7075529/build//../prebuilt/windows-x86_64/bin/echo.exe [x86] "Gdbsetup       ": "D:\Android\TestProjects\Git\shadowsocks-android\core\build\intermediates\cxx\Debug\202y2hp6/lib/x86/gdb.setup"
D:/Android/Sdk/ndk/21.4.7075529/build//../prebuilt/windows-x86_64/bin/echo.exe "set solib-search-path D:\Android\TestProjects\Git\shadowsocks-android\core\build\intermediates\cxx\Debug\202y2hp6/obj/local/x86" > D:\Android\TestProjects\Git\shadowsocks-android\core\build\intermediates\cxx\Debug\202y2hp6/lib/x86/gdb.setup
D:/Android/Sdk/ndk/21.4.7075529/build//../prebuilt/windows-x86_64/bin/echo.exe "directory D:/Android/Sdk/ndk/21.4.7075529/build//../toolchains/llvm/prebuilt/windows-x86_64/sysroot D:/Android/TestProjects/Git/shadowsocks-android/core/src/main/jni D:/Android/Sdk/ndk/21.4.7075529/build//../sources/android/support D:/Android/Sdk/ndk/21.4.7075529/build//../sources/cxx-stl/llvm-libc++ D:/Android/Sdk/ndk/21.4.7075529/build//../sources/cxx-stl/llvm-libc++abi/include D:/Android/TestProjects/Git/shadowsocks-android/core/src/main/jni/libancillary D:/Android/TestProjects/Git/shadowsocks-android/core/src/main/jni/badvpn/lwip/src/include/ipv4 D:/Android/TestProjects/Git/shadowsocks-android/core/src/main/jni/badvpn/lwip/src/include/ipv6 D:/Android/TestProjects/Git/shadowsocks-android/core/src/main/jni/badvpn/lwip/custom" >> D:\Android\TestProjects\Git\shadowsocks-android\core\build\intermediates\cxx\Debug\202y2hp6/lib/x86/gdb.setup

make: *** No rule to make target 'D:/Android/TestProjects/Git/shadowsocks-android/core/src/main/jni/redsocks/base.c', needed by 'D:\Android\TestProjects\Git\shadowsocks-android\core\build\intermediates\cxx\Debug\202y2hp6/obj/local/x86/objs-debug/redsocks/redsocks/base.o'.  Stop.

**Comment by patialvikas** [comment]
NDK version issue

**Reply by lsj19920916** [reply]
你好，你这边用的NDK版本是多少?我用的是21.4.7075529，依旧是这个问题

**Reply by lsj19920916** [reply]
Hello, what is the NDK version you are using? I used 21.4.7075529, which is still the same problem

**Comment by dev4u** [comment]
需要更多的信息

**Comment by hodakl099** [comment]
any info please?
